### ISSUE TITLE
addition of encoding parameter broke backward compatibility for Python versions < 3.9

### ISSUE THREAD

**Issue by richbon75** [Parquery/reconnecting-ftp#9]
Hi.  Nice library.  I wanted to use this in a project of ours that runs in Python 3.8, but I ran into a problem because Python only added support for the `encoding` argument for the [ftplib.FTP constructor in 3.9](https://docs.python.org/3/library/ftplib.html).  I noticed the [update to add support for the `encoding` argument](https://github.com/Parquery/reconnecting-ftp/pull/5) was very recent, so I'm [submitting a pull request](https://github.com/Parquery/reconnecting-ftp/pull/8) to fix the problem for versions of Python < 3.9.

Basically, it just detects if the `encoding` argument is causing a problem, and fails back to a version of the FTP constructor without the `encoding` argument.

Of note, I couldn't run your development tests as documented in the README.md, so you might want to make sure your instructions are up-to-date.  However, I can confirm that this code change is working fine for me in our project.

**Comment by mristin** [comment]
Thanks a lot for spotting this!

**Comment by mristin** [comment]
(pardon, wrong button)

**Comment by mristin** [comment]
@richbon75 I've just published a new patch version on PyPI. Could you please check that the newest version works correctly for you? If it works, feel free to close the issue unless there are more problems.

**Comment by richbon75** [comment]
Sweet!  Tested version 1.1.1 and it's working great.  Closing this issue.  Thanks very much for this library.  It's much better than what I was doing before, and a really clean drop-in replacement that seems to "just work".
